processor/stream/package_tests/span_attrs_test.go

```diff
@@ -0,0 +1,174 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package package_tests
+
+import (
+	"testing"
+
+	"github.com/elastic/apm-server/model/span/generated/schema"
+
+	"github.com/elastic/apm-server/processor/stream"
+	"github.com/elastic/apm-server/tests"
+)
+
+func spanProcSetup() *tests.ProcessorSetup {
+	return &tests.ProcessorSetup{
+		Proc:            &V2TestProcessor{StreamProcessor: stream.StreamProcessor{}},
+		FullPayloadPath: "../testdata/intake-v2/spans.ndjson",
+		Schema:          schema.ModelSchema,
+		SchemaPrefix:    "span",
+		TemplatePaths: []string{
+			"../../../model/span/_meta/fields.yml",
+			"../../../_meta/fields.common.yml",
+		},
+	}
+}
+
+func spanPayloadAttrsNotInFields() *tests.Set {
+	return tests.NewSet(
+		"span.stacktrace",
+		tests.Group("transaction.marks."),
+		tests.Group("context.db"),
+		"context.http",
+		"context.http.url",
+		"context.tags.tag1",
+	)
+}
+
+func spanFieldsNotInPayloadAttrs() *tests.Set {
+	return tests.Union(
+		tests.NewSet(
+			"listening",
+			"view spans",
+			"span.parent", // from v1
+		),
+		// not valid for the span context
+		transactionContext(),
+	)
+
+}
+
+func spanPayloadAttrsNotInJsonSchema() *tests.Set {
+	return tests.NewSet(
+		"span",
+		"span.context.request.headers.some-other-header",
+		"span.context.request.headers.array",
+		"span.stacktrace.vars.key",
+		"span.context.tags.tag1",
+		tests.Group("metadata"),
+	)
+}
+
+func spanJsonSchemaNotInPayloadAttrs() *tests.Set {
+	return tests.NewSet(
+		"span.transaction_id",
+	)
+}
+
+func spanRequiredKeys() *tests.Set {
+	return tests.NewSet(
+		"span",
+		"span.name",
+		"span.trace_id",
+		"span.parent_id",
+		"span.transaction_id",
+		"span.id",
+		"span.duration",
+		"span.type",
+		"span.start",
+		"span.stacktrace.filename",
+		"span.stacktrace.lineno",
+		"span.context.request.method",
+		"span.context.request.url",
+	)
+}
+
+func transactionContext() *tests.Set {
+	return tests.NewSet(
+		tests.Group("context.request.url"),
```

Comment:
you can remove `tests.Group("context.request.url")`, as it is covered by `tests.Group("context.request")`